pkgs/by-name/bi/binaryninja-free/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  unzip,
+  autoPatchelfHook,
+  dbus,
+  fontconfig,
+  freetype,
+  libGLU,
+  libxkbcommon,
+  wayland,
+  xcbutilimage,
+  xcbutilkeysyms,
+  xcbutilrenderutil,
+  xcbutilwm,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "binaryninja-free";
+  version = "4.2.6455";
+
+  src = fetchurl {
+    url = "https://web.archive.org/web/20241209150225/https://cdn.binary.ninja/installers/binaryninja_free_linux.zip";
+    hash = "sha256-NOVuLmko8iYcJ/0fr0DNw7xPEC8EhT/SzcFWtNmjlYI=";
+  };
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    dbus
+    fontconfig
+    freetype
+    libGLU
+    libxkbcommon
+    stdenv.cc.cc.lib
+    wayland
+    xcbutilimage
+    xcbutilkeysyms
+    xcbutilrenderutil
+    xcbutilwm
+  ];
+
```

Comment:
It would be nice to have a `.desktop` file for binary ninja (see [linux-setup.sh](https://github.com/Vector35/binaryninja-api/blob/7323628022239278956504476997716316149930/scripts/linux-setup.sh) for the desktop file generation they use in the personal version)

For example:

```nix
  nativeBuildInputs = [
    # .....
    copyDesktopItems
  ];

  icon = fetchurl {
    url = "https://raw.githubusercontent.com/Vector35/binaryninja-api/448f40be71dffa86a6581c3696627ccc1bdf74f2/docs/img/logo.png";
    hash = "sha256-TzGAAefTknnOBj70IHe64D6VwRKqIDpL4+o9kTw0Mn4=";
  };

  desktopItems = [
    (makeDesktopItem {
      name = "com.vector35.binaryninja";
      desktopName = "Binary Ninja Free";
      comment = "A Reverse Engineering Platform";
      exec = "binaryninja";
      icon = "binaryninja";
      mimeTypes = [ "application/x-binaryninja" "x-scheme-handler/binaryninja" ];
      categories = [ "Utility" ];
    })
  ];

  postInstall = ''
    # .....

    install -Dm644 ${icon} $out/share/icons/hicolor/256x256/apps/binaryninja.png

    runHook postInstall
  '';
```